avail-subxt/examples/submit_block_length_proposal.rs

```diff
@@ -0,0 +1,236 @@
+	tx_asend!(client, &vote, signer).map_err(Into::into)
+}
+
+/// Sets the block dimensions to default
+pub async fn reset(client: &Client, signer: &PairSigner) -> Result<()> {
+	println!("Resetting block dimensions for further tests");
+	let block_length_update = Call::DataAvailability(DaCall::submit_block_length_proposal {
+		rows: 256,
+		cols: 256,
+	});
+	let sudo_call = api::tx().sudo().sudo(block_length_update);
+	tx_send!(client, &sudo_call, signer);
+	Ok(())
+}
+
+/// Make `n` data submission transaction
+pub async fn submit_data(client: &Client, signer: &PairSigner, n: u8) -> Result<()> {
+	let example_data = b"X".repeat(1000).to_vec();
+	let data_submission = api::tx()
+		.data_availability()
+		.submit_data(BoundedVec(example_data));
+	for _ in 0..n {
+		tx_asend!(client, &data_submission, signer)?;
+	}
+
+	Ok(())
+}
+
+/// This example submits transactions to reduce the block dimensions
+/// This should work only if the block is mostly empty (using batch or not)
+#[async_std::main]
+async fn main() -> Result<()> {
+	let args = Opts::from_args();
+	let client = build_client(args.ws, args.validate_codegen).await?;
+	let signer = PairSigner::new(AccountKeyring::Alice.pair());
+
+	reset(&client, &signer).await?;
+
+	// Success cases
+	simple_tx(&client, &signer).await?;
+	batch_tx(&client, &signer).await?;
+	// tech_committee_tx(&client, &signer).await?;
+
+	// Fail cases
+	fail_simple_tx(&client, &signer).await?;
+	fail_batch_tx(&client, &signer).await?;
+
+	Ok(())
+}
+
+/** Success cases **/
+pub async fn simple_tx(client: &Client, signer: &PairSigner) -> Result<()> {
+	println!("1 - Sudo call to reduce the dimensions of the block.");
+	let block_length_update = Call::DataAvailability(DaCall::submit_block_length_proposal {
+		rows: BLOCK_DIM_VALUE,
+		cols: BLOCK_DIM_VALUE,
+	});
+	let sudo_call = api::tx().sudo().sudo(block_length_update);
+	tx_send!(client, &sudo_call, signer)
+		.find_first::<DaEvent::BlockLengthProposalSubmitted>()?
+		.expect("1 - Block Length Proposal Submitted event is emitted .qed");
+	println!("1 - Block Length Proposal Submitted found.");
+	reset(client, signer).await?;
+	Ok(())
+}
+
+pub async fn batch_tx(client: &Client, signer: &PairSigner) -> Result<()> {
+	println!("2 - Sudo call in a batch to reduce the dimensions of the block.");
+	let block_length_update = Call::DataAvailability(DaCall::submit_block_length_proposal {
+		rows: BLOCK_DIM_VALUE,
+		cols: BLOCK_DIM_VALUE,
+	});
+	let sudo_call = Call::Sudo(SudoCall::sudo {
+		call: Box::new(block_length_update),
+	});
+	let batch_call = api::tx().utility().batch(vec![sudo_call]);
+	tx_send!(client, &batch_call, signer)
+		.find_first::<DaEvent::BlockLengthProposalSubmitted>()?
+		.expect("2 - Block Length Proposal Submitted event is emitted .qed");
+	println!("2 - Block Length Proposal Submitted found.");
+	reset(client, signer).await?;
+	Ok(())
+}
+
+// TODO: See if submit_block_length_proposal needs to work with TC, Council and/or democracy
```

Comment:
Here, we don't use the tech committee test because in the current state, the `submit_block_length_proposal` is not allowed to process in tech committee / council / democracy / ... (only sudo)